include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
Addressed in 602d461: I believe this makes sure null checks for pointers passed to `h2o_memcpy` are not optimized away _and_ that we will crash if passed a `NULL` pointer with non-zero length.